tests/mentorship_relation/test_dao_creation.py

```diff
@@ -249,6 +250,40 @@ def test_dao_create_mentorship_relation_with_good_args_but_invalid_timestamp(sel
         self.assertEqual(messages.INVALID_END_DATE, result[0])
         self.assertEqual(HTTPStatus.BAD_REQUEST, result[1])
 
+    def test_dao_create_mentorship_relation_with_mentor_id_same_as_mentee_id(self):
+        dao = MentorshipRelationDAO()
+        data = dict(
+            mentor_id=1234,
+            mentee_id=1234,
+            end_date=self.end_date_example.timestamp(),
+            notes=self.notes_example,
+            tasks_list=TasksListModel(),
+        )
+        result = dao.create_mentorship_relation(1234, data)
+
+        self.assertEqual((messages.MENTOR_ID_SAME_AS_MENTEE_ID,400), result)
+
+        query_mentorship_relation = MentorshipRelationModel.query.first()
+
+        self.assertIsNone(query_mentorship_relation)
+    # def test_dao_create_mentorship_relation_with_end_time_before_present(self):
+        
+    #     dao = MentorshipRelationDAO()
+    #     data = dict(
+    #         mentor_id=self.first_user.id,
+    #         mentee_id=self.second_user.id,
+    #         now_datetime = datetime.now(),
+    #         end_date = datetime.fromtimestamp(data["end_date"]),
```

Comment:
From where you are taking `data[end_date]`. data is not yet created and you are trying to access it. :monocle_face: